src/doctor/__init__.py

```diff
@@ -54,14 +55,31 @@ class Rule(ABC):
       optional field. If no field is given, the value of the "message"
       field will be used.
 
+    dependencies: Dictionary with dependencies on packages and what
+    versions that are required.
+
     """
 
+    def get_versions(self, conn):
+        """Get extension versions."""
+        with conn.cursor() as cursor:
+            cursor.execute("SELECT extname, extversion FROM pg_extension WHERE extname IN %s",
+                           (tuple(self.dependencies.keys()),)) # pylint: disable=E1101
+            return {row[0]:row[1] for row in cursor}
```

Comment:
These need to be dictionary accesses, because of the usage of `RealDictCursor`
```suggestion
            return {row["extname"]: row["extversion"] for row in cursor}
```

Reply:
Well, it still supports [by-position indexing](https://www.psycopg.org/docs/extras.html), but it's more robust using a string key.

Reply:
I don't think that it does, from the [documentation](https://www.psycopg.org/docs/extras.html#psycopg2.extras.RealDictCursor):

> Note that this cursor is extremely specialized and does not allow the normal access (using integer indices) to fetched data. If you need to access database rows both as a dictionary and a list, then use the generic [DictCursor](https://www.psycopg.org/docs/extras.html#psycopg2.extras.DictCursor) instead of RealDictCursor.